tests/ui/invalid_pyclass_args.stderr

```diff
@@ -1,217 +1,282 @@
-  --> tests/ui/invalid_pyclass_args.rs:21:24
+  --> tests/ui/invalid_pyclass_args.rs:22:24
    |
-21 | #[pyclass(rename_all = "Camel-Case")]
+22 | #[pyclass(rename_all = "Camel-Case")]
    |                        ^^^^^^^^^^^^
 
 error: expected string literal
-  --> tests/ui/invalid_pyclass_args.rs:24:20
+  --> tests/ui/invalid_pyclass_args.rs:25:20
    |
-24 | #[pyclass(module = my_module)]
+25 | #[pyclass(module = my_module)]
    |                    ^^^^^^^^^
 
-error: expected one of: `crate`, `dict`, `eq`, `eq_int`, `extends`, `freelist`, `frozen`, `get_all`, `hash`, `mapping`, `module`, `name`, `rename_all`, `sequence`, `set_all`, `subclass`, `unsendable`, `weakref`
-  --> tests/ui/invalid_pyclass_args.rs:27:11
+error: expected one of: `crate`, `dict`, `eq`, `eq_int`, `extends`, `freelist`, `frozen`, `get_all`, `hash`, `mapping`, `module`, `name`, `rename_all`, `sequence`, `set_all`, `str`, `subclass`, `unsendable`, `weakref`
+  --> tests/ui/invalid_pyclass_args.rs:28:11
    |
-27 | #[pyclass(weakrev)]
+28 | #[pyclass(weakrev)]
    |           ^^^^^^^
 
 error: a `#[pyclass]` cannot be both a `mapping` and a `sequence`
-  --> tests/ui/invalid_pyclass_args.rs:31:8
+  --> tests/ui/invalid_pyclass_args.rs:32:8
    |
-31 | struct CannotBeMappingAndSequence {}
+32 | struct CannotBeMappingAndSequence {}
    |        ^^^^^^^^^^^^^^^^^^^^^^^^^^
 
 error: `eq_int` can only be used on simple enums.
-  --> tests/ui/invalid_pyclass_args.rs:52:11
+  --> tests/ui/invalid_pyclass_args.rs:53:11
    |
-52 | #[pyclass(eq_int)]
+53 | #[pyclass(eq_int)]
    |           ^^^^^^
 
 error: The `hash` option requires the `frozen` option.
-  --> tests/ui/invalid_pyclass_args.rs:59:11
+  --> tests/ui/invalid_pyclass_args.rs:60:11
    |
-59 | #[pyclass(hash)]
+60 | #[pyclass(hash)]
    |           ^^^^
 
 error: The `hash` option requires the `eq` option.
-  --> tests/ui/invalid_pyclass_args.rs:59:11
+  --> tests/ui/invalid_pyclass_args.rs:60:11
    |
-59 | #[pyclass(hash)]
+60 | #[pyclass(hash)]
    |           ^^^^
 
+error: invalid format string: expected `'}'` but string was terminated
+  --> tests/ui/invalid_pyclass_args.rs:93:1
+   |
+93 | #[pyclass(str = "{")]
+   | ^^^^^^^^^^^^^^^^^^^^^ expected `'}'` in format string
+   |
+   = note: if you intended to print `{`, you can escape it using `{{`
+   = note: this error originates in the attribute macro `pyclass` (in Nightly builds, run with -Z macro-backtrace for more info)
+
+error: invalid format string: expected `'}'`, found `'$'`
+  --> tests/ui/invalid_pyclass_args.rs:97:1
+   |
+97 | #[pyclass(str = "{$}")]
+   | ^^^^^^^^^^^^^^^^^^^^^^^ expected `'}'` in format string
+   |
+   = note: if you intended to print `{`, you can escape it using `{{`
+   = note: this error originates in the attribute macro `pyclass` (in Nightly builds, run with -Z macro-backtrace for more info)
+
```

Comment:
Thank you for the feedback!  While investigating this, I found an error in the value I was passing to to the quote macro for the formatting string.   I was passing the value instead of the LitStr, so the span was lost.  Now those errors you are highlighted above are [better scoped](https://github.com/PyO3/pyo3/pull/4233/files#diff-56de370817aae751f21fa633db392e620b3c619d009d3aa0b9f961df444ce11aR79-R98).  While this helps, I am going to add another compile error test for incorrect names or indices in the format string, where it would be nice to isolate the specific members within the string.  It looks like subspan might help with that, but I'm still trying to decipher its usage in practice.

Reply:
Sorted it out!  Here is output on nightly:
```
error: invalid format string: expected `'}'` but string was terminated
  --> tests/ui/invalid_pyclass_args.rs:93:19
   |
93 | #[pyclass(str = "{")]
   |                  -^ expected `'}'` in format string
   |                  |
   |                  because of this opening brace
   |
   = note: if you intended to print `{`, you can escape it using `{{`

error: invalid format string: expected `'}'`, found `'$'`
  --> tests/ui/invalid_pyclass_args.rs:97:19
   |
97 | #[pyclass(str = "{$}")]
   |                  -^ expected `'}'` in format string
   |                  |
   |                  because of this opening brace
   |
   = note: if you intended to print `{`, you can escape it using `{{`

error[E0609]: no field `aaaa` on type `&Point`
   --> tests/ui/invalid_pyclass_args.rs:101:22
    |
101 | #[pyclass(str = "X: {aaaa}, Y: {y}, Z: {z}")]
    |                      ^^^^ unknown field
    |
    = note: available fields are: `x`, `y`, `z`

error[E0609]: no field `zzz` on type `&Point2`
   --> tests/ui/invalid_pyclass_args.rs:109:40
    |
109 | #[pyclass(str = "X: {x}, Y: {y}}}, Z: {zzz}")]
    |                                        ^^^ unknown field
    |
    = note: available fields are: `x`, `y`, `z`
┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈

```
I will push the changes shortly.  The output is definitely nicer on nightly! :-)